pkg/controller/account/account_controller.go

```diff
@@ -660,35 +672,51 @@ func CreateAccount(client awsclient.Client, accountName, accountEmail string) (*
 }
 
 // CreateIAMUser takes a client and string and creates a IAMuser
-func CreateIAMUser(client awsclient.Client, userName string) (*iam.CreateUserOutput, error) {
+func CreateIAMUser(reqLogger logr.Logger, client awsclient.Client, userName string) (*iam.CreateUserOutput, error) {
 
 	// check if username exists for this account
 	_, err := client.GetUser(&iam.GetUserInput{
 		UserName: aws.String(userName),
 	})
 
-	awserr, ok := err.(awserr.Error)
-
-	if err != nil && awserr.Code() != iam.ErrCodeNoSuchEntityException {
-		return &iam.CreateUserOutput{}, err
-	}
-
 	var createUserOutput *iam.CreateUserOutput
-	if ok && awserr.Code() == iam.ErrCodeNoSuchEntityException {
-		createResult, err := client.CreateUser(&iam.CreateUserInput{
-			UserName: aws.String(userName),
-		})
-		if err != nil {
-			return &iam.CreateUserOutput{}, err
+	// handle errors
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
```

Comment:
should we return an error if this line is !ok ?

Reply:
Good catch, yes we should